BitwardenShared/Core/Auth/Services/AuthService.swift

```diff
@@ -594,18 +637,18 @@ class DefaultAuthService: AuthService { // swiftlint:disable:this type_body_leng
                 enforceOnLogin: model.enforceOnLogin ?? false
             )
         }
-        if try await requirePasswordChange(email: username, masterPassword: masterPassword, policy: policy) {
-            try await stateService.setForcePasswordResetReason(.weakMasterPasswordOnLogin)
-        }
-
-        if isNewAccount, await configService.getFeatureFlag(.nativeCreateAccountFlow) {
-            do {
-                let isAutofillEnabled = await credentialIdentityStore.isAutofillEnabled()
-                try await stateService.setAccountSetupAutofill(isAutofillEnabled ? .complete : .incomplete)
-                try await stateService.setAccountSetupImportLogins(.incomplete)
-                try await stateService.setAccountSetupVaultUnlock(.incomplete)
-            } catch {
-                errorReporter.log(error: error)
+        if try await requirePasswordChange(
+            email: username,
+            masterPassword: masterPassword,
+            preAuth: preAuth,
+            policy: policy
+        ) {
+            if preAuth {
+                // Since this is pre authentication we use a local var to cache this info.
+                forcePasswordResetReason = .weakMasterPasswordOnLogin
```

Comment:
🤔 Given this is cached in the service and this is a Singleton, could it happen that the user enters with master password then this cache is set and in the 2FA view the user goes back and changes account and when logging in it applies this reset reason to a different account? In the StateService this is already taken care of because it's tied to the `userId`.